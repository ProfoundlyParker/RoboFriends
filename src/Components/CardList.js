import React from "react";
import Card from './Card';

//Maps through JSON API to get robot's info
const CardList = ({ robots }) => {
    return (
    <div>
        {robots.map((robot) => {
                return (
                <Card 
                key={robot.id}
                firstName={robot.firstName} 
                lastName={robot.lastName}
                email={robot.email}
                image={robot.image}
                />
                );
            })}
        </div>
        );
    };

export default CardList;
