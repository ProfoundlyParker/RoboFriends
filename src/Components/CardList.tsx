import React from "react";
import Card from "./Card.tsx";

// Added robot info types
type Robot = {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  image: string;
};

// Allow robots to be an array of Robot, null, or undefined
type CardListProps = {
  robots: Robot[] | null | undefined;
};

//Maps through JSON API array to get robot's info
const CardList: React.FC<CardListProps> = ({ robots }) => {
  if (!Array.isArray(robots)) {
    return <div>Waiting for API...</div>;
  }
  return (
    <div>
      {robots.map((robot) => {
        return (
          <Card
            key={robot.id}
            firstName={robot.firstName}
            lastName={robot.lastName}
            email={robot.email}
            image={robot.image}
          />
        );
      })}
    </div>
  );
};

export default CardList;
