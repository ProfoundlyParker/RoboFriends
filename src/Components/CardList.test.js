import React from "react";
import { shallow } from "enzyme";
import CardList from "./CardList";

it("expect to snapshot CardList component", () => {
  const mockRobots = [
    {
      id: 1,
      name: "John Snow",
      username: "Johnny123",
      email: "john@gmail.com",
    },
  ];
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
