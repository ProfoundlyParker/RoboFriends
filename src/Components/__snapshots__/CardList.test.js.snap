// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to snapshot CardList component 1`] = `
<div>
  <Card
    email="john@gmail.com"
    key="1"
  />
</div>
`;
